janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/GhostVertexRemover.java

```diff
@@ -110,11 +110,18 @@ public void process(StaticBuffer key, Map<SliceQuery, EntryList> entries, ScanMe
             iterator.remove();
             removedRelations++;
         }
+        v.remove();
         //There should be no more system relations to remove
         metrics.incrementCustom(REMOVED_VERTEX_COUNT);
         metrics.incrementCustom(REMOVED_RELATION_COUNT,removedRelations);
     }
 
+    @Override
+    public void workerIterationEnd(final ScanMetrics metrics) {
+        tx.commit();
```

Comment:
Wouldn't it be better to commit in the `process` method? That way, the commit is closer to the mutation and I think that it also makes more sense to have those in the same method.
The documentation for the base method also says:

> Can be used to close any resources held by this job.

which means that I wouldn't expect a commit here.

Reply:
Process invoked multiple times for same tx so if we close transaction in intermediate state second process failed.

In the same doc:

> This method is semantically aligned with {@link org.apache.tinkerpop.gremlin.process.computer.VertexProgram#workerIterationEnd(Memory)}

And `workIterationEnd` has implementation that commit tx `BulkLoaderVertexProgram`